core/state_transition.go

```diff
@@ -149,6 +138,18 @@ func NewStateTransition(evm *vm.EVM, msg Message, gp *GasPool) *StateTransition
 // indicates a core error meaning that the message would always fail for that particular
 // state and would never be accepted within a block.
 func ApplyMessage(evm *vm.EVM, msg Message, gp *GasPool) ([]byte, uint64, bool, error) {
+	if vm.UsingOVM {
+		// OVM_ENABLED
+		ovmmsg, err := toExecutionManagerRun(evm, msg)
+		if err != nil {
+			return nil, 0, false, err
+		}
+
+		evm.Context.Origin = msg.From()
+
+		return NewStateTransition(evm, ovmmsg, gp).TransitionDb()
+	}
+
 	return NewStateTransition(evm, msg, gp).TransitionDb()
 }
```

Comment:
Believe you could do this instead?

```suggestion
	if vm.UsingOVM {
		// OVM_ENABLED
		msg, err := toExecutionManagerRun(evm, msg)
		if err != nil {
			return nil, 0, false, err
		}
		evm.Context.Origin = msg.From()
	}	

	return NewStateTransition(evm, msg, gp).TransitionDb()
}
```